src/main.js

```diff
@@ -70,6 +70,10 @@ module.exports = {
       }
     }))
 
+    this.subscriptions.add(atom.config.observe('linter-eslint.rulesToSilenceWhileTyping', (ids) => {
+      this.ignoredRulesWhenModified = idsToIgnoredRules(ids)
+    }))
```

Comment:
I'm trying to get the spec working, and finding it very difficult.  The core of the problem is that this subscription is what determines which rules are ignored when the file is modified.  In our current specs, this subscription is never added. When I try to call `linter.activate()` so the subscription is added, I get an error from the first line of the method which calls `require('atom-package-deps').install()`.  If I comment that out, the specs seem to work fine except that changing the config to add a rule to `rulesToSilenceWhileTyping` still is not observed (no idea why not) and therefore `this.ignoredRulesWhenModified` is never changed.  

So, long-and-short, I'm a bit stumped on how to proceed here.  I will push up a change that at least gets this rebased on master and makes the spec run correctly, but it will be failing.  Maybe someone else can take it from there.